catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -358,40 +359,39 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
                         "Found more metacards than updated metacards provided. Please ensure your attribute values match unique records.");
             }
 
-        } else if (pendingNrtIndex.getAllPresent(identifiers).size() > 0) {
-            Map<String, Metacard> matches = pendingNrtIndex.getAllPresent(identifiers);
-            for (Entry<String, Metacard> match : matches.entrySet()) {
-                idToMetacardMap.put(match.getKey(), match.getValue());
+            for (SolrDocument doc : idResults.getResults()) {
+                Metacard old;
+                try {
+                    old = client.createMetacard(doc);
+                } catch (MetacardCreationException e) {
+                    throw new IngestException("Could not create metacard(s).");
```

Comment:
This message seems technically correct, but ambiguous/confusing in the context of an update request. Shouldn't we at least rethrow the `MetacardCreationException`, or log the stacktrace and rethrow the message?